src/main/java/com/imsweb/geocoder/component/ProcessingPanel.java

```diff
@@ -98,6 +107,184 @@ public String toString() {
     private boolean _reachedEndOfFile = false;
     private boolean _cancelled = false;
 
+    // Penalty Code lookups    
+    private static List<Map<Character, String>> _penaltyCodeLookups = new ArrayList<>();
```

Comment:
This is very minor, but the Swing classes tend to become pretty big, and it's good practice to try to keep they short and focused. Could you please add a PenaltyCodeUtils class and move all these maps and the related utility methods into that class? That will keep thinks cleaner. Thanks.